modules/environment-roles/templates/tdr_jenkins_read_params.json.tpl

```diff
@@ -0,0 +1,22 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Action": "ssm:DescribeParameters",
+      "Resource": "*"
+    },
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Action": "ssm:GetParameters",
+      "Resource": [
+        "arn:aws:ssm:eu-west-2:${account_id}:parameter/${environment}/keycloak/admin/*",
+        "arn:aws:ssm:eu-west-2:${account_id}:parameter/${environment}/keycloak/client/secret",
+        "arn:aws:ssm:eu-west-2:${account_id}:parameter/${environment}/keycloak/backend_checks_client/secret",
+        "arn:aws:ssm:eu-west-2:${account_id}:parameter/${environment}/keycloak/realm_admin_client/secret"
```

Comment:
What does Jenkins use all of these parameters for? I'd have expected it to just need the last one.

Reply:
It needs the secrets values so they can be set in the configuration JSON at container startup and on updating. The configuration JSON does not contain the secret values when it is exported.

Reply:
[As above](https://github.com/nationalarchives/tdr-terraform-backend/pull/69#discussion_r467744669).